internal/printer/print-to-tsx.go

```diff
@@ -15,13 +15,17 @@ import (
 	"golang.org/x/net/html/atom"
 )
 
+func getTSXPrefix() string {
+	return "/** @jsx astroHTML */\n\n"
+}
+
 func PrintToTSX(sourcetext string, n *Node, opts transform.TransformOptions, h *handler.Handler) PrintResult {
 	p := &printer{
 		sourcetext: sourcetext,
 		opts:       opts,
 		builder:    sourcemap.MakeChunkBuilder(nil, sourcemap.GenerateLineOffsetTables(sourcetext, len(strings.Split(sourcetext, "\n")))),
 	}
-
+	p.print(getTSXPrefix())
```

Comment:
If you move this inside of `renderTsx()` > `if n.Type == DocumentNode`, right after `hasChildren := false`, do you still get the weird behavior?

Reply:
Same behaviour, yeah. That's actually where I put it originally, ha!